scrapy/utils/conf.py

```diff
@@ -13,7 +15,7 @@ def build_component_list(base, custom):
     """
     if isinstance(custom, (list, tuple)):
         return custom
-    compdict = base.copy()
+    compdict = BaseSettings(base, priority = 'default')
```

Comment:
I definitely try to stick to PEP8, but I have to admit that I frequently break the "no spaces for keyword arguments" rule b/c it feels so unnatural to me. Will fix :)